### ISSUE TITLE
Skip Wireguard virtual interfaces

### ISSUE THREAD

**Issue by Digitalone1** [AlynxZhou/gnome-shell-extension-net-speed#8]
Hello, thanks for this extension. I made a fork for myself since I like speeds in bits/s rather than bytes/s.

Anyway I saw you skip virtual interfaces, but unfortunately mine is not skipped with a VPN I use. I made a workaround here: https://github.com/Digitalone1/gnome-shell-extension-net-speed/commit/ffc33940725fc5b82bf7f5710cf0111c31e48753

Just added `iface.startsWith("wg-")`. Maybe you want to add it to your branch.

And maybe for performance reasons it would be better to make one only aggregated regex. See here: https://regex101.com/r/v1ByIq/1

Bye.

**Comment by AlynxZhou** [comment]
> Just added iface.startsWith("wg-"). Maybe you want to add it to your branch.

On my system wireguard looks like `wg0`, not `wg-0`, is this a typo?

**Comment by AlynxZhou** [comment]
> And maybe for performance reasons it would be better to make one only aggregated regex. See here: https://regex101.com/r/v1ByIq/1

Sounds great, and I think use `test()` is enough, we never need `match()`'s result.

**Comment by Digitalone1** [comment]
> > Just added iface.startsWith("wg-"). Maybe you want to add it to your branch.
> 
> On my system wireguard looks like `wg0`, not `wg-0`, is this a typo?

No, you can make a regex like that: https://regex101.com/r/3Ae2be/1

**Comment by AlynxZhou** [comment]
> No, you can make a regex like that: https://regex101.com/r/3Ae2be/1

OK, you mean it could be `wg123` or `wg-`, not `wg-123`, right?

**Comment by AlynxZhou** [comment]
I think the following is OK:

```
iface === "lo" ||
// `ifb`: Created by python-based bandwidth manager "traffictoll".
// `lxdbr`: Created by lxd container manager.
/^(?:ifb|lxdbr|virbr|br|vnet|tun|tap|docker|utun)\d+/.test(iface) ||
/^(?:wg)(\d+|-)/.test(iface) ||
// Skip interfaces like `veth1dd2116@if5`.
iface.startsWith("veth")
```

**Comment by Digitalone1** [comment]
> OK, you mean it could be `wg123` or `wg-`, not `wg-123`, right?

Mine it's `wg-nameofthevpn`, but it would skip also `wg-123`

> I think the following is OK:
> 
> ```
> iface === "lo" ||
> // `ifb`: Created by python-based bandwidth manager "traffictoll".
> // `lxdbr`: Created by lxd container manager.
> /^(?:ifb|lxdbr|virbr|br|vnet|tun|tap|docker|utun)\d+/.test(iface) ||
> /^(?:wg)(\d+|-)/.test(iface) ||
> // Skip interfaces like `veth1dd2116@if5`.
> iface.startsWith("veth")
> ```

If you use groups, make them non-capturing with `?:`. And you don't need `+` quantifier on the number, just skip on the first one.

Also, you could parse the code with [prettier](https://archlinux.org/packages/extra/any/prettier/). Just run `prettier extension.js --write`.

**Comment by AlynxZhou** [comment]
Actually I am thinking about use `startsWith()` for all interface names, looks like they all uses prefix.

**Comment by AlynxZhou** [comment]
Closed with <https://github.com/AlynxZhou/gnome-shell-extension-net-speed/commit/cf25936e571ea7d8f6acd19251f5b706af531ce6>.

**Comment by AlynxZhou** [comment]
> Closed with [cf25936](https://github.com/AlynxZhou/gnome-shell-extension-net-speed/commit/cf25936e571ea7d8f6acd19251f5b706af531ce6).

Oops. Left a typo in that commit, please pull the latest of master branch.

**Comment by Digitalone1** [comment]
> > Closed with [cf25936](https://github.com/AlynxZhou/gnome-shell-extension-net-speed/commit/cf25936e571ea7d8f6acd19251f5b706af531ce6).
> 
> Oops. Left a typo in that commit, please pull the latest of master branch.

Don't worry, I will keep it different because I only need to show the speeds in bits. Maybe I will only merge the method to check the virtual interface. Thank you anyway.
